### PR TITLE
Feature:(issue_1215) Allow calling commands by shorthand

### LINKED ISSUE #1215: Allow shorthands by calling commands via a unique name substring
## Checklist

* [x] Are you running the latest v2 release? The list of releases is [here](https://github.com/urfave/cli/releases).
* [x] Did you check the manual for your release? The v2 manual is [here](https://github.com/urfave/cli/blob/master/docs/v2/manual.md)
* [x] Did you perform a search about this feature? Here's the [Github guide](https://help.github.com/en/github/managing-your-work-on-github/using-search-to-filter-issues-and-pull-requests) about searching.

## What problem does this solve?
Command aliases are somewhat limited for shortening command names.

## Solution description
Instead of explicit aliases, commands could be called via a part of their name only, as long as that is unique among all commands.
The `ip` utility or the network manager cli (`nmcli`) implement this behaviour and give a good showcase of the usability enhancements this brings:
Peek 2020-12-16 12-30
To avoid ambiguity between subcommands & and command arguments, this behaviour should be disableable per command, eg via a setting `cmd.AcceptsArguments`

### PR DESCRIPTION
**Pull request by dearchap**
## What type of PR is this?

_(REQUIRED)_

- feature

## What this PR does / why we need it:

_(REQUIRED)_

Allows closest match calling of command 

## Which issue(s) this PR fixes:

_(REQUIRED)_

Fixes #1215 

## Special notes for your reviewer:

_(fill-in or delete this section)_

## Testing

_(fill-in or delete this section)_

## Release Notes

_(REQUIRED)_

```release-note

```

### CODE REVIEW

**Review by meatballhat — COMMENTED** [review]
How about instead of `AllowShorthandCommand` going with `PrefixMatchCommands`?

**Comment by dearchap** [comment]
@meatballhat Renamed

File: command.go [hunk 1]
```diff
@@ -64,6 +64,9 @@ type Command struct {
 	// render custom help text by setting this variable.
 	CustomHelpTemplate string
 
+	// Use longest prefix match for commands
+	AllowShorthandCommand bool
```
Comment (meatballhat) [inline command.go:68]: Rename this one to `PrefixMatchCommands`, too?

Comment (dearchap) [reply command.go:68]: Done

**Review by meatballhat — APPROVED** [review]
Exciting!

**Pull request merged**